internal/controller/controllers/agent_controller.go

```diff
@@ -193,32 +218,42 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 				},
 			})
 			if err2 != nil {
-				return r.updateStatus(ctx, log, agent, origAgent, &h.Host, nil, err2, !IsUserError(err2))
+				return r.updateStatus(ctx, log, agent, origAgent, &h.Host, nil, nil, err2, !IsUserError(err2))
 			}
-			return r.updateStatus(ctx, log, agent, origAgent, &host.Host, cluster.ID, nil, true)
+
+			return r.updateStatus(ctx, log, agent, origAgent, &host.Host, cluster.ID, cluster, nil, true)
 		} else if *h.ClusterID != *cluster.ID {
 			log.Infof("ClusterDeploymentName is changed in Agent %s. unbind first", agent.Name)
 			return r.unbindHost(ctx, log, agent, origAgent, h)
 		}
 	}
 
+	// Retrieve cluster by ClusterDeploymentName from the database
+	cluster, err2 := r.getCluster(agent)
```

Comment:
Why is it needed again ?

Reply:
The cluster that we retrieve on line 202 is out of scope by this point, or at the very least, you have no guarantee that cluster would be populated, so you have to reload it.

The reason cluster is being retrieved here is so that it can be passed to updateStatus where the cluster database object is interrogated for whether or not UserManagedNetworking is enabled.